crates/store/re_entity_db/src/entity_db.rs

```diff
@@ -657,6 +682,8 @@ impl EntityDb {
                 .write()
                 .cache()
                 .purge_fraction_of_ram(fraction_to_purge);
+        } else {
+            self.on_store_events(&store_events);
```

Comment:
I'm confused about all the additional stuff that are happening now through this call. Why did we get away without it so far? Are we doing this twice for some reason now since all the handling already happens on `self.gc`? 🤔 
Conceptually it makes a lot of sense to me. There's events and we should react to them.

Reply:
There was a bug in the old code. We should have called it here, but we didn't.